scylla/src/statement/prepared_statement.rs

```diff
@@ -134,9 +137,10 @@ impl PreparedStatement {
     /// [Self::calculate_token()].
     pub fn compute_partition_key(
         &self,
-        bound_values: &SerializedValues,
+        bound_values: &impl SerializeRow,
```

Comment:
This used to take `SerializedValues` as an argument, why does it take `impl SerializeRow` now? Shouldn't it rather take `NewSerializedValues` instead? This should remove the need to add `PartitionKeyError::Serialization` variant.

Reply:
This is a method on `PreparedStatement` and I wanted it to be type safe - it couldn't with `NewSerializedValues`.
Also, `NewSerializedValues` are not convenient for a user, it's much easier to call this method with `SerializeRow`.

Reply:
OK, makes sense.